### PR TITLE
Test coverage

### PR DESCRIPTION
**Pull request by SoA-Lee**
### 변경사항
**AS-IS**
1. 회원 탈퇴 / 내 정보 확인
2. Authority에 따른 화면 설정 변경
3. DTO 변경을 통한 전체적인 코드 수정

**TO-BE**
1. 기능 구현
   - 상품 관련 Entity 작성하기
   - 상품 등록 및 수정하기
   - 관리자 기능 추가

2. 테스트코드 작성

### CODE REVIEW

**Comment by SoA-Lee** [comment]
|File|Coverage [14.11%]|:x:|
|:-|:-:|:-:|
|[MemberRole.java](https://github.com/SoA-Lee/N-BUY/blob/f54fb7b7fa234bf178a36aafd7635c5550de510c/src%2Fmain%2Fjava%2Fcom%2Fdalgorithm%2Fnbuy%2Fmember%2Fentity%2FMemberRole.java)|100%|:green_apple:|
|[ErrorCode.java](https://github.com/SoA-Lee/N-BUY/blob/f54fb7b7fa234bf178a36aafd7635c5550de510c/src%2Fmain%2Fjava%2Fcom%2Fdalgorithm%2Fnbuy%2Fexception%2FErrorCode.java)|97.67%|:green_apple:|
|[MemberServiceImpl.java](https://github.com/SoA-Lee/N-BUY/blob/f54fb7b7fa234bf178a36aafd7635c5550de510c/src%2Fmain%2Fjava%2Fcom%2Fdalgorithm%2Fnbuy%2Fmember%2Fservice%2Fimpl%2FMemberServiceImpl.java)|21.23%|:x:|
|[MemberException.java](https://github.com/SoA-Lee/N-BUY/blob/f54fb7b7fa234bf178a36aafd7635c5550de510c/src%2Fmain%2Fjava%2Fcom%2Fdalgorithm%2Fnbuy%2Fmember%2Fexception%2FMemberException.java)|16.67%|:x:|
|[Member.java](https://github.com/SoA-Lee/N-BUY/blob/f54fb7b7fa234bf178a36aafd7635c5550de510c/src%2Fmain%2Fjava%2Fcom%2Fdalgorithm%2Fnbuy%2Fmember%2Fentity%2FMember.java)|11.02%|:x:|
|[MemberFormDto.java](https://github.com/SoA-Lee/N-BUY/blob/f54fb7b7fa234bf178a36aafd7635c5550de510c/src%2Fmain%2Fjava%2Fcom%2Fdalgorithm%2Fnbuy%2Fmember%2Fdto%2FMemberFormDto.java)|0%|:x:|
|[MemberDto.java](https://github.com/SoA-Lee/N-BUY/blob/f54fb7b7fa234bf178a36aafd7635c5550de510c/src%2Fmain%2Fjava%2Fcom%2Fdalgorithm%2Fnbuy%2Fmember%2Fdto%2FMemberDto.java)|0%|:x:|
|[UserAuthenticationFailureHandler.java](https://github.com/SoA-Lee/N-BUY/blob/f54fb7b7fa234bf178a36aafd7635c5550de510c/src%2Fmain%2Fjava%2Fcom%2Fdalgorithm%2Fnbuy%2Fconfiguration%2FUserAuthenticationFailureHandler.java)|0%|:x:|
|[SecurityConfiguration.java](https://github.com/SoA-Lee/N-BUY/blob/f54fb7b7fa234bf178a36aafd7635c5550de510c/src%2Fmain%2Fjava%2Fcom%2Fdalgorithm%2Fnbuy%2Fconfiguration%2FSecurityConfiguration.java)|0%|:x:|
|[UserDetailsServiceImpl.java](https://github.com/SoA-Lee/N-BUY/blob/f54fb7b7fa234bf178a36aafd7635c5550de510c/src%2Fmain%2Fjava%2Fcom%2Fdalgorithm%2Fnbuy%2Fmember%2Fservice%2Fimpl%2FUserDetailsServiceImpl.java)|0%|:x:|
|[PasswordUtils.java](https://github.com/SoA-Lee/N-BUY/blob/f54fb7b7fa234bf178a36aafd7635c5550de510c/src%2Fmain%2Fjava%2Fcom%2Fdalgorithm%2Fnbuy%2Fmember%2Futil%2FPasswordUtils.java)|0%|:x:|
|[AdminMainController.java](https://github.com/SoA-Lee/N-BUY/blob/f54fb7b7fa234bf178a36aafd7635c5550de510c/src%2Fmain%2Fjava%2Fcom%2Fdalgorithm%2Fnbuy%2Fadmin%2FAdminMainController.java)|0%|:x:|
|[MemberController.java](https://github.com/SoA-Lee/N-BUY/blob/f54fb7b7fa234bf178a36aafd7635c5550de510c/src%2Fmain%2Fjava%2Fcom%2Fdalgorithm%2Fnbuy%2Fmember%2Fcontroller%2FMemberController.java)|0%|:x:|
|[GlobalExceptionHandler.java](https://github.com/SoA-Lee/N-BUY/blob/f54fb7b7fa234bf178a36aafd7635c5550de510c/src%2Fmain%2Fjava%2Fcom%2Fdalgorithm%2Fnbuy%2Fexception%2FGlobalExceptionHandler.java)|0%|:x:|

|Total Project Coverage|10.84%|:x:|
|:-|:-:|:-:|

File: src/main/java/com/dalgorithm/nbuy/member/entity/Member.java [hunk 1]
```diff
@@ -28,12 +33,33 @@ public class Member implements MemberCode {
...
+        member.setUserId(memberFormDto.getUserId());
+        member.setUserName(memberFormDto.getUserName());
+        member.setUserEmail(memberFormDto.getUserEmail());
+        member.setPhone(memberFormDto.getPhone());
+        member.setPassword(enPassword);
+        member.setRole(MemberRole.ROLE_USER);
+        member.setRegDt(LocalDateTime.now());
+        member.setEmailAuthYn(false);
+        member.setEmailAuthKey(uuid);
+
```
Comment (devycha) [inline src/main/java/com/dalgorithm/nbuy/member/entity/Member.java:62]: 멤버필드가 꽤 많은데 builder 말고 setter로 모두 구현하신 특별한 이유가 혹시 있으신가요??

File: src/main/java/com/dalgorithm/nbuy/configuration/SecurityConfiguration.java [hunk 2]
```diff
@@ -29,17 +34,18 @@ UserAuthenticationFailureHandler getFailureHandler() {
...
         // 페이지 권한 설정 - 주소 매칭 (아래 주소에 대해 접근 가능하게 처리)
         http.authorizeRequests()
                 .antMatchers(
                         "/"
-                        , "/member/register"
-                        , "/member/email_auth"
+                        , "/members/register"
+                        , "/members/email_auth"
                 )
                 .permitAll();
```
Comment (shinyurim) [inline src/main/java/com/dalgorithm/nbuy/configuration/SecurityConfiguration.java:50]: 소아님 Role에 Admin, User 2개가 있는데 사용자와 관리자 둘 다  접근가능한 페이지 / 관리자만 접근 가능한 페이지도 나누어 넣어도 좋을 것 같아요..!

http.authorizeRequests()
                .antMatchers("/admin/**).hasRole("ROLE_ADMIN) // 관리자만 접근 가능한 사이트

이런 식으로 /admin으로 시작하는 부분은 관리자만 접근 가능하고 일반 사용자는 접근 못하게 막는 것도 좋을 것 같습니닷

File: src/main/java/com/dalgorithm/nbuy/member/controller/MemberController.java [hunk 3]
```diff
@@ -23,24 +26,37 @@ public class MemberController {
 
     private final MemberService memberService;
 
-    @RequestMapping("/member/login")
+    @RequestMapping("/members/login")
```
Comment (shinyurim) [inline src/main/java/com/dalgorithm/nbuy/member/controller/MemberController.java:29]: Controller 맵핑 주소들에 공통적으로 "/members" 가 들어가 있는데,
public class MemberContoller{} 위에 @RequestMapping("/members") 이런 식으로 작성해주시면 맵핑 부분에서는 /member 생략 가능하더라구요! 저는 이게 편해서 이렇게 쓰고 있는데, 한 번 사용해 보시는 것도 좋을 것 같습니다~!

**Pull request merged**